src/poptconfig.c

```diff
@@ -136,13 +136,17 @@ int poptReadFile(const char * fn, char ** bp, size_t * nbp, int flags)
 	goto exit;
 
     if ((nb = lseek(fdno, 0, SEEK_END)) == (off_t)-1
+     || (uintmax_t)nb >= SIZE_MAX
      || lseek(fdno, 0, SEEK_SET) == (off_t)-1
      || (b = calloc(sizeof(*b), (size_t)nb + 1)) == NULL
      || read(fdno, (char *)b, (size_t)nb) != (ssize_t)nb)
     {
 	int oerrno = errno;
 	(void) close(fdno);
-	errno = oerrno;
+	if (nb != (off_t)-1 && (uintmax_t)nb >= SIZE_MAX)
+	    errno = -EOVERFLOW;
```

Comment:
Wouldn't EFBIG (file too large) be more descriptive?

Reply:
Yeah, it was my first idea as well. But then I checked what open(2) would do in such a case. As described in the manual page, POSIX.1 states that EOVERFLOW has to be returned. Linux before 2.6.24 returned EFBIG, so I chose to stick with POSIX.

Reply:
Oh, I see. And actually in POSIX, EFBIG seems to be used to signal *writing* a file too big.
Fair enough then, thanks for the patch & attention to detail :+1: